accepted/2020/mono-convergence/globalization-support-in-.net5.md

```diff
@@ -0,0 +1,90 @@
+
+### iOS/tvOS
+There is no build in support for ICU on apple mobile platforms. For our purposes of having highly compatible globalization mode across all platform we support we need to look into bringing ICU to ios/tvOS as well.
+
+System.Globalization.Invariant mode still need to work without any ICU dependency. How we are actually going to slice the actual data will be defined later based on the experience and existing ICU tooling.
+
+We are as expected not the first framework to hit this problem. Swift framework is dealing with same problems and they seem to prefer a solution which drop ICU dependency but that still does not look like final [decision](https://forums.swift.org/t/icu-usage-in-swift/20473).
+
+### WebAssembly
+The intention is set invariant globalization mode as the default mode for everyone for WebAssembly but at the same time allow a fallback to full globalization support when needed. This will come with additional cost but on the web platform we have little bit more flexibility how to deploy and share the data.
+
+ICU library consists from code and data. What we need to explore is if there is reasonable way to package the ICU data and store them in the cloud as a publicly available data blobs. This could give us flexibility to support full range of ICU data on demand if sliced into small data sets. We’ll still need to ship icu implementation but that could be also bound dynamically using dynamic linking. 
+
+From our earlier ICU work, we do know that libicuuc and libicui18n are dynamically linked to libicudata. We would need to download and make libicudata available before loading ICU. 
+
+## ICU Packaging
+Due to lack of ICU on all platform we intend to support for .NET5 we need to look into how to package it for following RIDs
+
+* ios-x64
+* ios-arm
+* ios-arm64
+* tvos-x64
+* tvos-arm64
+* webassembly (browser mode)
+
+A default build of ICU normally results in over 16 MB of data, and a substantial amount of object code.
+
+The packages have to be split to implementation and data part and ideally the data parts will be also sliced in the way that the developers can select only relevant data for their app if they for example don’t target Chinese market.
+
+### Data Library Customizer (TBD)
+There is a [tool](http://www.icu-project.org/docs/demo/datacustom_help.html) built by ICU which allow customizing what kind of data will be included in the final data set. We need to explore if that something we could leverage either for the data islands or even as something using during build process.
+
+Additional documentation about the tool is [here](https://github.com/unicode-org/icu/blob/master/docs/userguide/icu_data/buildtool.md)
+
+## App Local ICU for Windows
+Windows teams is working on packaging ICU for .NET needs. It will be available as NuGet package and we could explore to use a similar mechanism for iOS/tvOS packaging (although Windows will contain the whole data file). 
```

Comment:
```suggestion
The Windows teams is working on packaging ICU for .NET needs. It will be available as a NuGet package, and we could explore using a similar mechanism for iOS/tvOS packaging (although Windows will contain the whole data file). 
```